ens/registrar.py

```diff
@@ -0,0 +1,245 @@
+        labels = [self._to_label(label) for label in labels]
+        label_hashes = [self.ens.labelhash(label) for label in labels]
+        return self.core.startAuctions(label_hashes, **modifier_dict)
+
+    def bid(self, label, amount, secret, **modifier_dict):
+        """
+        :param str label: to bid on
+        :param int amount: wei to bid
+        :param secret: You **MUST keep a copy of this** to avoid burning your entire deposit!
+        :type secret: bytes, str or int
+        """
+        if not modifier_dict:
+            modifier_dict = {'transact': {}}
+        if 'transact' in modifier_dict:
+            transact = modifier_dict['transact']
+            self.__default_gas(transact, 'bid')
+            if 'value' not in transact:
+                transact['value'] = amount
+            elif transact['value'] < amount:
+                raise UnderfundedBid("Bid of %s ETH was only funded with %s ETH" % (
+                                     Web3().fromWei(amount, 'ether'),
+                                     Web3().fromWei(transact['value'], 'ether')))
+        label = self._to_label(label)
+        sender = self.__require_sender(modifier_dict)
+        if amount < MIN_BID:
+            raise BidTooLow("You must bid at least %s ether" % Web3().fromWei(MIN_BID, 'ether'))
+        bid_hash = self._bid_hash(label, sender, amount, secret)
+        return self.core.newBid(bid_hash, **modifier_dict)
+
+    def reveal(self, label, amount, secret, **modifier_dict):
+        if not modifier_dict:
+            modifier_dict = {'transact': {}}
+        if 'transact' in modifier_dict:
+            self.__default_gas(modifier_dict['transact'], 'reveal')
+        sender = self.__require_sender(modifier_dict)
+        label = self._to_label(label)
+        bid_hash = self._bid_hash(label, sender, amount, secret)
+        if not self.core.sealedBids(sender, bid_hash):
+            raise InvalidBidHash
+        label_hash = self.ens.labelhash(label)
+        secret_hash = self._secret_hash(secret)
+        return self.core.unsealBid(label_hash, amount, secret_hash, **modifier_dict)
+    unseal = reveal
+
+    def finalize(self, label, **modifier_dict):
+        if not modifier_dict:
+            modifier_dict = {'transact': {}}
+        if 'transact' in modifier_dict:
+            self.__default_gas(modifier_dict['transact'], 'finalize')
+        label = self._to_label(label)
+        label_hash = self.ens.labelhash(label)
+        return self.core.finalizeAuction(label_hash, **modifier_dict)
+
+    def entries_by_hash(self, label_hash):
+        assert isinstance(label_hash, (bytes, bytearray))
+        entries = self.core.entries(label_hash)
+        return AuctionEntries(
+            Status(entries[0]),
+            self._deedContract(entries[1]) if entries[1] else None,
+            datetime.fromtimestamp(entries[2], pytz.utc) if entries[2] else None,
```

Comment:
does `datetime.datetime.utcfromtimestamp`? give us the same thing here and eliminate the need for `pytz`?